src/registry/parseRegistry.spec.ts

```diff
@@ -46,4 +46,586 @@ describe('parseRegistry', () => {
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'westend',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly update the registry with an injectedRegsitry without tokens while ignoring specName', () => {
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'0': {
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+						specName: 'totoro',
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['0']).toStrictEqual({
+			tokens: ['WND', 'WND2'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'westend',
+			poolPairsInfo: {},
+		});
+	});
+	it("Should correctly update the registry's token entry with an injectedRegsitry", () => {
+		const assetsInfo = {};
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'0': {
+						tokens: ['WOP'],
+						assetsInfo,
+						foreignAssetsInfo,
+						poolPairsInfo: {},
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['0']).toStrictEqual({
+			tokens: ['WND', 'WND2', 'WOP'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'westend',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly add a previously missing injectedRegsitry without assetsInfo', () => {
+		const foreignAssetsInfo = {};
+		const opts = {
+			injectedRegistry: {
+				rococo: {
+					'2000': {
+						foreignAssetsInfo,
+						tokens: ['TST'],
+						poolPairsInfo: {},
+						specName: 'testy',
+					},
+				},
+			},
+		};
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.rococo['2000']).toStrictEqual({
+			tokens: ['TST'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'testy',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly add an injectedRegistry only with tokens and specName', () => {
+		const opts = {
+			injectedRegistry: {
+				westend: {
+					'4000': {
+						tokens: ['GRN'],
+						specName: 'guarani',
+					},
+				},
+			},
+		};
+
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.westend['4000']).toStrictEqual({
+			tokens: ['GRN'],
+			assetsInfo: {},
+			foreignAssetsInfo: {},
+			specName: 'guarani',
+			poolPairsInfo: {},
+		});
+	});
+	it('Should correctly ignore the existing entries in assetsInfo and tokens', () => {
+		const assetsInfo = {
+			'0': 'POTATO',
+			'1': 'L T',
+			'2': 'RANOVER',
+			'3': 'Meow',
+			'4': 'HAPPY',
+			'2511': 'TESTY',
+			'3611': 'MATE',
+		};
+		const tokens = ['KSM'];
+		const opts = {
+			injectedRegistry: {
+				kusama: {
+					'1000': {
+						tokens: tokens,
+						assetsInfo: assetsInfo,
+					},
+				},
+			},
+		};
+
+		const registry = parseRegistry(reg as ChainInfoRegistry, opts);
+
+		expect(registry.kusama['1000'].assetsInfo).toStrictEqual({
```

Comment:
This kind of check is unmaintainable, since the registry assets will grow, and we will have to update this test everytime we update the package. Instead just see if the value exists. Same with the tests below.